source/Ship.cpp

```diff
@@ -4401,9 +4401,20 @@ double Ship::CalculateDeterrence() const
 			const Outfit *weapon = hardpoint.GetOutfit();
 			if(weapon->Ammo() && weapon->AmmoUsage() && !OutfitCount(weapon->Ammo()))
 				continue;
-			double strength = weapon->ShieldDamage() + weapon->HullDamage()
-				+ (weapon->RelativeShieldDamage() * attributes.Get("shields"))
-				+ (weapon->RelativeHullDamage() * attributes.Get("hull"));
+			// Ignore disabled and asteroid damages.
+			// Effects over time are considered over 60 frames, and disruption is % based so multiply it by 100 too.
+			// First consider special effects, then hull, shileds, energy and heat.
+			double strength = weapon->DisruptionDamage() * 600.
+				+ weapon->SlowingDamage() * 60.
+				+ (weapon->FuelDamage() + weapon->LeakDamage() * 60)
+				+ (weapon->ShieldDamage() + weapon->RelativeShieldDamage() * attributes.Get("shields"))
+					+ weapon->DischargeDamage() * 60.
+				+ (weapon->HullDamage() + weapon->RelativeHullDamage() * attributes.Get("hull"))
+					+ weapon->CorrosionDamage() * 60.
+				+ ((weapon->EnergyDamage() + weapon->RelativeEnergyDamage() * attributes.Get("energy capacity"))
+					+ weapon->IonDamage() * 60. + weapon->ScramblingDamage() * 60) / 2.
+				+ ((weapon->HeatDamage() + weapon->RelativeHeatDamage() * MaximumHeat())
+					+ weapon->BurnDamage() * 60.) / 10.;
```

Comment:
We're effectively giving every damage type a similar rating with these multipliers, which I don't think makes much sense. For example, fuel damage can be a deterrent, but how often will fuel damage actually influence the outcome of a battle? Extremely few ships use fuel damage weapons.

I think that the primary means of deterrence should be shield and hull damage. If we include special damage types, then the influence that those special damage types have on the deterrence strength should be lower.

Reply:
I would consider fuel damage to be a deterrent, not because it's likely to be the difference between victory and defeat, but because it's likely to be the difference between escape and destruction. The ability to eliminate a ship's ability to escape is a serious threat, even if it's unlikely to directly influence a hypothetical duel.

Reply:
well the influence of those damage types is lower already for all weapons pretty much - heat gets divided by 10 because its easy to get a big amount of that, but for other types 100 slowing is a lot? I did this with different weapons in mind, trying to make them have an appropriate amount of deterrance but we can tweak it for sure